ballerina/tests/hub_with_readonly_params_test.bal

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+import ballerina/mime;
+import ballerina/test;
+
+service /websubhub on new Listener(9102) {
+    isolated remote function onRegisterTopic(readonly & TopicRegistration msg) returns TopicRegistrationSuccess {
+        test:assertTrue(msg is readonly);
+        return TOPIC_REGISTRATION_SUCCESS;
+    }
+
+    isolated remote function onDeregisterTopic(readonly & TopicDeregistration msg) returns TopicDeregistrationSuccess {
+        test:assertTrue(msg is readonly);
+        return TOPIC_DEREGISTRATION_SUCCESS;
+    }
+
+    isolated remote function onUpdateMessage(readonly & UpdateMessage msg) returns Acknowledgement {
+        test:assertTrue(msg is readonly);
+        return ACKNOWLEDGEMENT;
+    }
+
+    isolated function onSubscription(readonly & Subscription msg) returns SubscriptionAccepted {
+        test:assertTrue(msg is readonly);
+        return SUBSCRIPTION_ACCEPTED;
+    }
+
+    remote function onSubscriptionValidation(readonly & Subscription msg) {
+        test:assertTrue(msg is readonly);
+    }
+    
+    isolated remote function onSubscriptionIntentVerified(readonly & VerifiedSubscription msg) {
+        test:assertTrue(msg is readonly);
+    }
+
+    isolated function onUnsubscription(readonly & Subscription msg) returns SubscriptionAccepted {
+        return UNSUBSCRIPTION_ACCEPTED;
```

Comment:
Don't we need to assert `msg is readonly` here?

Reply:
Fixed with [b250ffe997a8e72e2c0f30040e0682634f350ac2](https://github.com/ballerina-platform/module-ballerina-websubhub/pull/424/commits/b250ffe997a8e72e2c0f30040e0682634f350ac2)